### DISCUSSION TITLE
Granular UWP* application control

### DISCUSSION THREAD

**Discussion by ghost** [wiresock/proxifyre#10] [Ideas]
It pains me to create this issue, for I have scouted the web all day long in the quest for The Perfect Socksifier. Proxifier was good, yet it lacked what I needed most: UDP support. Win2Socks refused to start on my server entirely, and ProxyCap, WideCap and SocksCap64 never had a chance of functioning decades after they were conceived.

Now that I have stumbled upon your creation, I am humbly asking you to touch it up a little, as I would like to get it to work with a beloved UWP application of mine called ROBLOX. I have attempted to add Windows10Universal to my appNames array, but unsurprisingly, no traffic was being proxied. What has worked, instead, was adding svchost to the whitelist; I managed to launch the game successfully. My VPN client's split tunneling feature exhibits a similar behaviour.

I will be using multiple ROBLOX clients at once, and need each of them to use a separate proxy; aside from the svchost limiting me, I will also encounter issues with the way process names are handled; there will be multiple executables with the same name but in different directories. I would like to be able to specify the file path, but I have tried with C:\\Windows\\System32\\curl.exe and it didn't seem to be picked up by ProxyFyre.

Thanks again for the wonderful tool, I hope you are doing well.

**Comment by wiresock** [comment]
Thank you for your interest in ProxiFyre! 
I've conducted tests with ROBLOX, and ProxiFyre recognizes the associated application as:
`Process ID: 29788`
`Path: C:\Program Files\WindowsApps\ROBLOXCORPORATION.ROBLOX_2.590.678.0_x86__55nm5eh3cm0pr\Windows10Universal.exe`

In version 1.0.16, I've made adjustments so that if a forward slash or backslash is present in the pattern, it matches against the full pathname. This approach mirrors the one used for the Wiresock VPN Client.

I've verified the following configuration with both ROBLOX and Firefox. It effectively routes Firefox and ROBLOX over distinct proxies:

```json
[
    {
        "appNames": ["firefox"],
        "socks5ProxyEndpoint": "127.0.0.1:8080"
    },
    {
        "appNames": ["C:\\Program Files\\WindowsApps\\ROBLOXCORPORATION.ROBLOX_2.590.678.0_x86__55nm5eh3cm0pr"],
        "socks5ProxyEndpoint": "IP_ADDRESS_HERE:1080",
        "username": "danteuser",
        "password": "PASSWORD_HERE"
    }
]
```
Kindly give it a test and share your feedback with me.

**Comment by ghost** [comment]
I have tested the update with a dozen game clients and and it works beautifully, thank you!
